pkg/scheduler/framework/v1alpha1/interface.go

```diff
@@ -86,6 +89,19 @@ func NewStatus(code Code, msg string) *Status {
 	}
 }
 
+type WaitingPod struct {
+	pod *v1.Pod
+	s   chan *Status
+}
+
+func (w *WaitingPod) GetPod() *v1.Pod {
+	return w.pod
+}
+
+func (w *WaitingPod) SetStatus(status *Status) {
```

Comment:
I am not sure about `SetStatus`. I am more inclined towards providing an `Approve` and a `Reject` function. Those functions must also be exposed by the `framework`.